includes/razorpay-subscription-webhook.php

```diff
@@ -199,8 +202,43 @@ protected function processSubscriptionSuccess($orderId, $subscription, $paymentI
      *
      * @param $orderId
      */
-    protected function processSubscriptionFailed($orderId)
+    protected function processSubscriptionFailed($orderId, $subscription, $paymentId)
     {
-        WC_Subscriptions_Manager::process_subscription_payment_failure_on_order($orderId);
+        $wcSubscription = wcs_get_subscriptions_for_order($orderId);
+        $wcSubscription = array_values($wcSubscription)[0];
+
+        $is_first_payment = ( $wcSubscription->get_completed_payment_count() < 1 ) ? true : false;
```

Comment:
ternary is not needed. `<` will return a boolean

Reply:
$wcSubscription->get_completed_payment_count() < 1  should be <=1.

Case authentication payment got success, 
 get_completed_payment_count gets updated as 1.
Now hook got triggered.  and renewal orders get updated

Reply:
@ccrims0n  One more minor fix. SInce the first payment is upfront amount.
The length of subscription should         $length = (int) WC_Subscriptions_Product::get_length($product['product_id']) -1 

Needs to add that check. Now we have paid for one extra sub.

Reply:
Also that payment could be paid via razorpay also